src/main/java/org/radarcns/webapp/resource/DataSetEndPoint.java

```diff
@@ -155,20 +151,24 @@ public Dataset getSamplesJson(
             @Alphanumeric @PathParam(SOURCE_ID) String sourceId,
             @Alphanumeric @PathParam(SOURCE_DATA_NAME) String sourceDataName,
             @PathParam(STAT) DescriptiveStatistic stat,
-            @PathParam(TIME_WINDOW) TimeWindow interval,
-            @QueryParam(START) Long start,
-            @QueryParam(END) Long end) throws IOException {
+            @QueryParam(TIME_WINDOW) TimeWindow interval,
+            @ISO8601DateString @QueryParam(START) String start,
```

Comment:
From https://jersey.github.io/documentation/latest/user-guide.html#d0e2270

> In general the Java type of the method parameter may:
> 
> 1. Be a primitive type;
> 
> 2. Have a constructor that accepts a single String argument;
> 
> 3. Have a static method named valueOf or fromString that accepts a single String argument (see, for example, Integer.valueOf(String) and java.util.UUID.fromString(String));
> 
> 4. Have a registered implementation of `javax.ws.rs.ext.ParamConverterProvider` JAX-RS extension SPI that returns a `javax.ws.rs.ext.ParamConverter` instance capable of a "from string" conversion for the type. or
> 
> 5. Be List<T>, Set<T> or SortedSet<T>, where T satisfies 2. or 3. above. The resulting collection is read-only.

I don't know if Jackson already does point 4, but otherwise I'd prefer that over having an annotation and having to do the conversion inline.

Reply:
I had a look. Both 2 and 4 are options. 2 is rather straight-forward. From https://jersey.github.io/documentation/latest/jaxrs-resources.html#d0e2271